In [30]:
import sklearn.datasets as skds
import numpy as np
import pandas as pd

In [2]:
dig = skds.load_digits()

In [3]:
bc = skds.load_breast_cancer()

In [22]:
dig.keys()

['images', 'data', 'target_names', 'DESCR', 'target']

In [27]:
dig['data'][0] 

array([  0.,   0.,   5.,  13.,   9.,   1.,   0.,   0.,   0.,   0.,  13.,
        15.,  10.,  15.,   5.,   0.,   0.,   3.,  15.,   2.,   0.,  11.,
         8.,   0.,   0.,   4.,  12.,   0.,   0.,   8.,   8.,   0.,   0.,
         5.,   8.,   0.,   0.,   9.,   8.,   0.,   0.,   4.,  11.,   0.,
         1.,  12.,   7.,   0.,   0.,   2.,  14.,   5.,  10.,  12.,   0.,
         0.,   0.,   0.,   6.,  13.,  10.,   0.,   0.,   0.])

In [33]:
dig['target'][:15]

array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4])

In [34]:
bc.keys()

['target_names', 'data', 'target', 'DESCR', 'feature_names']

In [36]:
bc['data'][0]

array([  1.79900000e+01,   1.03800000e+01,   1.22800000e+02,
         1.00100000e+03,   1.18400000e-01,   2.77600000e-01,
         3.00100000e-01,   1.47100000e-01,   2.41900000e-01,
         7.87100000e-02,   1.09500000e+00,   9.05300000e-01,
         8.58900000e+00,   1.53400000e+02,   6.39900000e-03,
         4.90400000e-02,   5.37300000e-02,   1.58700000e-02,
         3.00300000e-02,   6.19300000e-03,   2.53800000e+01,
         1.73300000e+01,   1.84600000e+02,   2.01900000e+03,
         1.62200000e-01,   6.65600000e-01,   7.11900000e-01,
         2.65400000e-01,   4.60100000e-01,   1.18900000e-01])

In [37]:
bc['target'][:15]

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])

In [39]:
from sklearn import cross_validation
from sklearn import naive_bayes

In [49]:
clf = naive_bayes.BernoulliNB()
bern_score_dig = cross_validation.cross_val_score(clf,dig['data'], dig['target'], scoring='accuracy').mean()
bern_score_bc = cross_validation.cross_val_score(clf,bc['data'], bc['target'], scoring='accuracy').mean()
print "Бернулли:", '\n', 'Датасет digits: %.3f \nДатасет breast cancer: %.3f' % (bern_score_dig , bern_score_bc)

Бернулли: 
Датасет digits: 0.826 
Датасет breast cancer: 0.627


In [50]:
clf = naive_bayes.MultinomialNB()
mult_score_dig = cross_validation.cross_val_score(clf,dig['data'], dig['target'], scoring='accuracy').mean()
mult_score_bc = cross_validation.cross_val_score(clf,bc['data'], bc['target'], scoring='accuracy').mean()
print "Мультиномиальное:", '\n', 'Датасет digits: %.3f \nДатасет breast cancer: %.3f' % (mult_score_dig , mult_score_bc)

Мультиномиальное: 
Датасет digits: 0.871 
Датасет breast cancer: 0.895


In [52]:
clf = naive_bayes.GaussianNB()
gaus_score_dig = cross_validation.cross_val_score(clf,dig['data'], dig['target'], scoring='accuracy').mean()
gaus_score_bc = cross_validation.cross_val_score(clf,bc['data'], bc['target'], scoring='accuracy').mean()
print "Гауссово:", '\n', 'Датасет digits: %.3f \nДатасет breast cancer: %.3f' % (gaus_score_dig , gaus_score_bc)

Гауссово: 
Датасет digits: 0.819 
Датасет breast cancer: 0.937


1) Максимальное качество на breast cancer: 0.937 (Гауссово распределение)

2) Максимальное качество на digits: 0.871  (мультиномиальное распределение)

3) Верные утверждения: d